gschem/src/x_image.c

```diff
@@ -129,7 +129,7 @@ x_image_get_type_from_description (const char *description)
   GSList *ptr;
 
   if (strcmp (description, "Portable Document Format") == 0) {
-    return "pdf";
+    return (char*) "pdf";
```

Comment:
I think the buffer returned by `x_image_get_type_from_description()` is not owned by the caller, and because the buffer should not be modified or free'd by the caller.  Maybe it should return a `const char*` rather than a `char*`?